From 3d62e95fc7cb7eb74d157dd62fb7e3ce607662ac Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Wed, 1 Jun 2016 21:38:22 +0100
Subject: [PATCH 3/5] mutt-1.5.23-ssl_ciphers

---
 conn/gnutls.c | 59 +++++++++++++++++++++++++++------------------------
 mutt_config.c |  2 +-
 2 files changed, 32 insertions(+), 29 deletions(-)

diff --git a/conn/gnutls.c b/conn/gnutls.c
index 3c93ba8b03..898fd232df 100644
--- a/conn/gnutls.c
+++ b/conn/gnutls.c
@@ -758,36 +758,39 @@ static int tls_set_priority(struct TlsSockData *data)
   else
     mutt_buffer_strcpy(priority, "NORMAL");
 
-  if (!C_SslUseTlsv13)
+  if (C_SslCiphers && (strcmp(C_SslCiphers, "@SYSTEM") == 0))
   {
-    nproto--;
-    mutt_buffer_addstr(priority, ":-VERS-TLS1.3");
-  }
-  if (!C_SslUseTlsv12)
-  {
-    nproto--;
-    mutt_buffer_addstr(priority, ":-VERS-TLS1.2");
-  }
-  if (!C_SslUseTlsv11)
-  {
-    nproto--;
-    mutt_buffer_addstr(priority, ":-VERS-TLS1.1");
-  }
-  if (!C_SslUseTlsv1)
-  {
-    nproto--;
-    mutt_buffer_addstr(priority, ":-VERS-TLS1.0");
-  }
-  if (!C_SslUseSslv3)
-  {
-    nproto--;
-    mutt_buffer_addstr(priority, ":-VERS-SSL3.0");
-  }
+    if (!C_SslUseTlsv13)
+    {
+      nproto--;
+      mutt_buffer_addstr(priority, ":-VERS-TLS1.3");
+    }
+    if (!C_SslUseTlsv12)
+    {
+      nproto--;
+      mutt_buffer_addstr(priority, ":-VERS-TLS1.2");
+    }
+    if (!C_SslUseTlsv11)
+    {
+      nproto--;
+      mutt_buffer_addstr(priority, ":-VERS-TLS1.1");
+    }
+    if (!C_SslUseTlsv1)
+    {
+      nproto--;
+      mutt_buffer_addstr(priority, ":-VERS-TLS1.0");
+    }
+    if (!C_SslUseSslv3)
+    {
+      nproto--;
+      mutt_buffer_addstr(priority, ":-VERS-SSL3.0");
+    }
 
-  if (nproto == 0)
-  {
-    mutt_error(_("All available protocols for TLS/SSL connection disabled"));
-    goto cleanup;
+    if (nproto == 0)
+    {
+      mutt_error(_("All available protocols for TLS/SSL connection disabled"));
+      goto cleanup;
+    }
   }
 
   int err = gnutls_priority_set_direct(data->state, mutt_b2s(priority), NULL);
diff --git a/mutt_config.c b/mutt_config.c
index fe82575d73..2cdd508800 100644
--- a/mutt_config.c
+++ b/mutt_config.c
@@ -4372,7 +4372,7 @@ struct ConfigDef MuttVars[] = {
   ** .te
   */
 #endif /* USE_SSL_GNUTLS */
-  { "ssl_ciphers", DT_STRING, &C_SslCiphers, 0 },
+  { "ssl_ciphers", DT_STRING, &C_SslCiphers, IP "@SYSTEM" },
   /*
   ** .pp
   ** Contains a colon-separated list of ciphers to use when using SSL.
