From 8fd83f9b96861e0b9363063cc51d4f4165abf4b5 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Wed, 1 Jun 2016 21:38:43 +0100
Subject: [PATCH 4/5] mutt-1.5.23-system_certs

---
 conn/config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conn/config.c b/conn/config.c
index 3e4afc211..c2dafe9c8 100644
--- a/conn/config.c
+++ b/conn/config.c
@@ -96,7 +96,7 @@ static struct ConfigDef ConnVarsSsl[] = {
 #if defined(USE_SSL_GNUTLS)
 static struct ConfigDef ConnVarsGnutls[] = {
   // clang-format off
-  { "ssl_ca_certificates_file", DT_PATH|DT_PATH_FILE, 0, 0, NULL,
+  { "ssl_ca_certificates_file", DT_PATH|DT_PATH_FILE, IP "/etc/ssl/certs/ca-bundle.crt", 0, NULL,
     "File containing trusted CA certificates"
   },
   { "ssl_min_dh_prime_bits", DT_NUMBER|DT_NOT_NEGATIVE, 0, 0, NULL,
