From 31018b98500b3d6a377a83441ad86157eea7d7f9 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Wed, 1 Jun 2016 21:38:22 +0100
Subject: [PATCH 3/5] mutt-1.5.23-ssl_ciphers

---
 conn/config.c |  2 +-
 conn/gnutls.c | 59 +++++++++++++++++++++++++++------------------------
 2 files changed, 32 insertions(+), 29 deletions(-)

diff --git a/conn/config.c b/conn/config.c
index 4c04eb456..e8f9c9486 100644
--- a/conn/config.c
+++ b/conn/config.c
@@ -92,7 +92,7 @@ struct ConfigDef ConnVars[] = {
     "File containing trusted CA certificates"
   },
 #endif
-  { "ssl_ciphers", DT_STRING, &C_SslCiphers, 0, 0, NULL,
+  { "ssl_ciphers", DT_STRING, &C_SslCiphers, IP "@SYSTEM", 0, NULL,
     "Ciphers to use when using SSL"
   },
   { "ssl_client_cert", DT_PATH|DT_PATH_FILE, &C_SslClientCert, 0, 0, NULL,
diff --git a/conn/gnutls.c b/conn/gnutls.c
index 1ae7ee3a2..b8f0404c1 100644
--- a/conn/gnutls.c
+++ b/conn/gnutls.c
@@ -761,36 +761,39 @@ static int tls_set_priority(struct TlsSockData *data)
   else
     mutt_buffer_strcpy(priority, "NORMAL");
 
-  if (!C_SslUseTlsv13)
+  if (C_SslCiphers && (strcmp(C_SslCiphers, "@SYSTEM") == 0))
   {
-    nproto--;
-    mutt_buffer_addstr(priority, ":-VERS-TLS1.3");
-  }
-  if (!C_SslUseTlsv12)
-  {
-    nproto--;
-    mutt_buffer_addstr(priority, ":-VERS-TLS1.2");
-  }
-  if (!C_SslUseTlsv11)
-  {
-    nproto--;
-    mutt_buffer_addstr(priority, ":-VERS-TLS1.1");
-  }
-  if (!C_SslUseTlsv1)
-  {
-    nproto--;
-    mutt_buffer_addstr(priority, ":-VERS-TLS1.0");
-  }
-  if (!C_SslUseSslv3)
-  {
-    nproto--;
-    mutt_buffer_addstr(priority, ":-VERS-SSL3.0");
-  }
+    if (!C_SslUseTlsv13)
+    {
+      nproto--;
+      mutt_buffer_addstr(priority, ":-VERS-TLS1.3");
+    }
+    if (!C_SslUseTlsv12)
+    {
+      nproto--;
+      mutt_buffer_addstr(priority, ":-VERS-TLS1.2");
+    }
+    if (!C_SslUseTlsv11)
+    {
+      nproto--;
+      mutt_buffer_addstr(priority, ":-VERS-TLS1.1");
+    }
+    if (!C_SslUseTlsv1)
+    {
+      nproto--;
+      mutt_buffer_addstr(priority, ":-VERS-TLS1.0");
+    }
+    if (!C_SslUseSslv3)
+    {
+      nproto--;
+      mutt_buffer_addstr(priority, ":-VERS-SSL3.0");
+    }
 
-  if (nproto == 0)
-  {
-    mutt_error(_("All available protocols for TLS/SSL connection disabled"));
-    goto cleanup;
+    if (nproto == 0)
+    {
+      mutt_error(_("All available protocols for TLS/SSL connection disabled"));
+      goto cleanup;
+    }
   }
 
   int err = gnutls_priority_set_direct(data->state, mutt_b2s(priority), NULL);
