diff --git c/doc/Muttrc.head w/doc/Muttrc.head
index 69e27ca..040c6f6 100644
--- c/doc/Muttrc.head
+++ w/doc/Muttrc.head
@@ -24,11 +24,15 @@ macro index,pager,attach,compose \cb "\
 
 # Show documentation when pressing F1
 macro generic,pager <F1> "<shell-escape> less @docdir@/manual.txt<Enter>" "show Mutt documentation"
+# and also F2, as some terminals use F1
+macro generic,pager <F2> "<shell-escape> less @docdir@/manual.txt<Enter>" "show Mutt documentation"
 
 # show the incoming mailboxes list (just like "mutt -y") and back when pressing "y"
 macro index,pager y "<change-folder>?<toggle-mailboxes>" "show incoming mailboxes list"
 bind browser y exit
 
+bind editor <delete> delete-char
+
 # Use folders which match on \\.gz$ as gzipped folders:
 # open-hook \\.gz$ "gzip -cd %f > %t"
 # close-hook \\.gz$ "gzip -c %t > %f"
