From 4d14ca674d494acd66367d1bed69068c3bc078b0 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Wed, 1 Jun 2016 21:37:32 +0100
Subject: [PATCH 1/5] mutt-1.5.18-muttrc

---
 doc/Muttrc.head | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/doc/Muttrc.head b/doc/Muttrc.head
index dd5b3c5..28e7b6b 100644
--- a/doc/Muttrc.head
+++ b/doc/Muttrc.head
@@ -24,6 +24,8 @@ macro index,pager,attach,compose \cb "\
 
 # Show documentation when pressing F1
 macro generic,pager <F1> "<shell-escape> less @docdir@/manual.txt<Enter>" "show Mutt documentation"
+# and also F2, as some terminals use F1
+macro generic,pager <F2> "<shell-escape> less @docdir@/manual.txt<Enter>" "show Mutt documentation"
 
 # show the incoming mailboxes list (just like "mutt -y") and back when pressing "y"
 macro index,pager y "<change-folder>?<toggle-mailboxes>" "show incoming mailboxes list"
@@ -34,6 +36,8 @@ bind browser y exit
 # close-hook  '\.gz$' "gzip --stdout              '%t' >  '%f'"
 # append-hook '\.gz$' "gzip --stdout              '%t' >> '%f'"
 
+bind editor <delete> delete-char
+
 # If Mutt is unable to determine your site's domain name correctly, you can
 # set the default here.
 #
-- 
2.7.4

