From b268f175cdf64b4c16f747f57905b145a6307ff2 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Wed, 1 Jun 2016 21:38:22 +0100
Subject: [PATCH 3/4] mutt-1.5.23-ssl_ciphers

---
 init.h            |  2 +-
 mutt_ssl_gnutls.c | 51 +++++++++++++++++++++++++++------------------------
 2 files changed, 28 insertions(+), 25 deletions(-)

diff --git a/init.h b/init.h
index f22c542c..42aaa3c3 100644
--- a/init.h
+++ b/init.h
@@ -3924,7 +3924,7 @@ struct Option MuttVars[] = {
   */
 #endif /* defined HAVE_SSL_PARTIAL_CHAIN */
 #endif /* defined USE_SSL_OPENSSL */
-  { "ssl_ciphers", DT_STRING, R_NONE, UL &SslCiphers, UL 0 },
+  { "ssl_ciphers", DT_STRING, R_NONE, UL &SslCiphers, UL "@SYSTEM" },
   /*
   ** .pp
   ** Contains a colon-seperated list of ciphers to use when using SSL.
diff --git a/mutt_ssl_gnutls.c b/mutt_ssl_gnutls.c
index 1ea0cda0..14344277 100644
--- a/mutt_ssl_gnutls.c
+++ b/mutt_ssl_gnutls.c
@@ -936,32 +936,35 @@ static int tls_set_priority(struct TlsSockData *data)
   else
     safe_strcat(priority, priority_size, "NORMAL");
 
-  if (!option(OPT_SSL_USE_TLSV1_2))
+  if (SslCiphers && strcmp(SslCiphers, "@SYSTEM"))
   {
-    nproto--;
-    safe_strcat(priority, priority_size, ":-VERS-TLS1.2");
-  }
-  if (!option(OPT_SSL_USE_TLSV1_1))
-  {
-    nproto--;
-    safe_strcat(priority, priority_size, ":-VERS-TLS1.1");
-  }
-  if (!option(OPT_SSL_USE_TLSV1))
-  {
-    nproto--;
-    safe_strcat(priority, priority_size, ":-VERS-TLS1.0");
-  }
-  if (!option(OPT_SSL_USE_SSLV3))
-  {
-    nproto--;
-    safe_strcat(priority, priority_size, ":-VERS-SSL3.0");
-  }
+    if (!option(OPT_SSL_USE_TLSV1_2))
+    {
+      nproto--;
+      safe_strcat(priority, priority_size, ":-VERS-TLS1.2");
+    }
+    if (!option(OPT_SSL_USE_TLSV1_1))
+    {
+      nproto--;
+      safe_strcat(priority, priority_size, ":-VERS-TLS1.1");
+    }
+    if (!option(OPT_SSL_USE_TLSV1))
+    {
+      nproto--;
+      safe_strcat(priority, priority_size, ":-VERS-TLS1.0");
+    }
+    if (!option(OPT_SSL_USE_SSLV3))
+    {
+      nproto--;
+      safe_strcat(priority, priority_size, ":-VERS-SSL3.0");
+    }
 
-  if (nproto == 0)
-  {
-    mutt_error(_("All available protocols for TLS/SSL connection disabled"));
-    FREE(&priority);
-    return -1;
+    if (nproto == 0)
+    {
+      mutt_error(_("All available protocols for TLS/SSL connection disabled"));
+      FREE(&priority);
+      return -1;
+    }
   }
 
   if ((err = gnutls_priority_set_direct(data->state, priority, NULL)) < 0)
