From a0f4b6366cb54c7cf163bbb276f120e77131e266 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Tue, 3 Dec 2019 15:58:06 +0000
Subject: [PATCH 5/5] neomutt-2019-rhel

---
 auto.def | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/auto.def b/auto.def
index 5aed53011..90bf8c3e9 100644
--- a/auto.def
+++ b/auto.def
@@ -433,10 +433,6 @@ if {[get-define want-gpgme]} {
       if {$gpgme_maj < 1 || $gpgme_min < 8} {
         user-error "Found GPGME version $gpgme_version, need 1.8.0 for AutoCrypt"
       }
-    } else {
-      if {$gpgme_maj < 1 || $gpgme_min < 4} {
-        user-error "Found GPGME version $gpgme_version, need 1.4.0"
-      }
     }
     define GPGME_VERSION_NUMBER [format "0x%02x%02x%02x" $gpgme_maj $gpgme_min $gpgme_patch]
 
